lib/ansiblelint/_prerun.py

```diff
@@ -39,6 +40,56 @@ def prepare_environment() -> None:
         os.environ['ANSIBLE_ROLES_PATH'] = "roles"
         print("Added ANSIBLE_ROLES_PATH=roles", file=sys.stderr)
 
+    if os.path.exists("requirements.yml"):
+
+        cmd = [
+            "ansible-galaxy",
+            "install",
+            "--roles-path",
+            ".cache/roles",
+            "-vr",
+            "requirements.yml"
+            ]
+
+        print("Running %s" % " ".join(cmd))
+        run = subprocess.run(
+            cmd,
+            universal_newlines=True,
+            check=False,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        if run.returncode != 0:
+            sys.exit(run.returncode)
+
+        cmd = [
+            "ansible-galaxy",
+            "collection",
+            "install",
+            "-p",
+            ".cache/collections",
+            "-vr",
+            "requirements.yml"
+            ]
+
+        print("Running %s" % " ".join(cmd))
+        run = subprocess.run(
+            cmd,
+            universal_newlines=True,
+            check=False,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        if run.returncode != 0:
+            sys.exit(run.returncode)
+
+        os.environ['ANSIBLE_ROLES_PATH'] = f".cache/roles:{os.environ['ANSIBLE_ROLES_PATH']}"
+        if 'ANSIBLE_COLLECTIONS_PATHS' in os.environ:
```

Comment:
Why is line 87 guarded, but line 86 is not? Is there a guarantee somewhere that ANSIBLE_ROLES_PATH exists?

Reply:
Fixed.